src/Bridge/Doctrine/Orm/Filter/SearchFilter.php

```diff
@@ -270,7 +270,7 @@ protected function filterProperty(string $property, $value, QueryBuilder $queryB
 
         if ($metadata->isCollectionValuedAssociation($association)) {
             $associationAlias = QueryBuilderHelper::addJoinOnce($queryBuilder, $queryNameGenerator, $alias, $association);
-            $associationField = 'id';
+            $associationField = ($metadata->getAssociationMapping($association)['joinTable']['inverseJoinColumns'][0]['referencedColumnName'] ?? 'id');
```

Comment:
Nice, this is indeed a long-time known bug, I'm not sure that the fix is that easy but if you make the tests pass that's already a good improvement!